### PR TITLE
Update bulk (mixture) density after mechanics sequential solve

### PR DESCRIPTION
**Pull request by paveltomin**
Main thing in this PR: update bulk (mixture) density after mechanics sequential solve. That way porosity changes are accounted in outer loop convergence check. It mostly matter for initialization, to match initial state with FIM.

Side thing: cleanup porosity update functions and make sure `FixedStress` suffix marks ones used in sequential mode.

https://github.com/GEOS-DEV/integratedTests/pull/99

### CODE REVIEW

File: src/coreComponents/constitutive/solid/PorousSolid.hpp [hunk 1]
```diff
@@ -152,6 +155,22 @@ class PorousSolidUpdates : public CoupledSolidUpdates< SOLID_TYPE, BiotPorosity,
     real64 const bulkModulus = m_solidUpdate.getBulkModulus( k );
     real64 const meanEffectiveStressIncrement = bulkModulus * ( strainIncrement[0] + strainIncrement[1] + strainIncrement[2] );
     m_porosityUpdate.updateMeanEffectiveStressIncrement( k, q, meanEffectiveStressIncrement );
+
+    // Compute porosity and its derivatives
+    real64 const deltaPressure = pressure - pressure_n;
+    real64 const deltaTemperature = temperature - temperature_n;
+    real64 porosityInit, dPorosity_dPressure, dPorosity_dTemperature;
+    computePorosity( k,
```
Comment (paveltomin) [inline src/coreComponents/constitutive/solid/PorousSolid.hpp]: @castelletto1 @joshua-white @CusiniM i am not sure if fixed stress version of porosity update should be used here?

**Review by jhuang2601 — APPROVED** [review]

**Comment by jhuang2601** [comment]
@paveltomin Based on `ci: run integrated tests`, following failure seems problematic
` FAIL RUN   : PoroElastic_hybridHexPrism_co2_sequential_3d_01             : poroElasticCoupling/PoroElastic_hybridHexPrism_co2_sequential_3d_01                : 0:05:51 : 0:05:51   : geos `

**Comment by paveltomin** [comment]
> @paveltomin Based on `ci: run integrated tests`, following failure seems problematic `FAIL RUN : PoroElastic_hybridHexPrism_co2_sequential_3d_01 : poroElasticCoupling/PoroElastic_hybridHexPrism_co2_sequential_3d_01 : 0:05:51 : 0:05:51 : geos`

thanks, i wish i can move it forward but this base PR https://github.com/GEOS-DEV/GEOS/pull/2866 needs to go first

File: src/coreComponents/physicsSolvers/solidMechanics/kernels/FixedStressThermoPoromechanics_impl.hpp [hunk 2]
```diff
@@ -172,7 +179,32 @@ quadraturePointKernel( localIndex const k,
                                                                          stack.uhat_local,
                                                                          reinterpret_cast< real64 (&)[numNodesPerElem][3] >(stack.localResidual),
                                                                          -stabilizationScaling );
-  stiffness.template upperBTDB< numNodesPerElem >( dNdX, -detJxW, stack.localJacobian );
+  stiffness.template BTDB< numNodesPerElem >( dNdX, -detJxW, stack.localJacobian );
```
Comment (paveltomin) [inline src/coreComponents/physicsSolvers/solidMechanics/kernels/FixedStressThermoPoromechanics_impl.hpp]: @castelletto1  @joshua-white @CusiniM somehow this way does not work well for this specific case https://github.com/GEOS-DEV/GEOS/blob/develop/inputFiles/poromechanics/PoroElastic_hybridHexPrism_co2_sequential_3d.xml
any idea why?

Comment (paveltomin) [reply src/coreComponents/physicsSolvers/solidMechanics/kernels/FixedStressThermoPoromechanics_impl.hpp]: @castelletto1 is this way legal https://github.com/GEOS-DEV/GEOS/blob/15f8d14e65c3f3e340bbf407b084f1585725f14c/src/coreComponents/physicsSolvers/solidMechanics/kernels/FixedStressThermoPoromechanics_impl.hpp#L182 ?

Comment (paveltomin) [reply src/coreComponents/physicsSolvers/solidMechanics/kernels/FixedStressThermoPoromechanics_impl.hpp]: the porosity derivative does not seem to work properly, see initialization step for sequential run below:
```
  Iteration  1: linearElasticity
    Attempt:  0, ConfigurationIter:  0, NewtonIter:  0
        ( Rsolid ) = ( 1.69e+01 )        ( R ) = ( 1.69e+01 )
        Last LinSolve(iter,res) = (   1, 2.88e-15 )
        linearElasticity: Global solution scaling factor = 1
    Attempt:  0, ConfigurationIter:  0, NewtonIter:  1
        ( Rsolid ) = ( 1.63e-02 )        ( R ) = ( 1.63e-02 )
        Last LinSolve(iter,res) = (   1, 6.88e-15 )
        linearElasticity: Global solution scaling factor = 1
    Attempt:  0, ConfigurationIter:  0, NewtonIter:  2
        ( Rsolid ) = ( 1.96e-05 )        ( R ) = ( 1.96e-05 )
        Last LinSolve(iter,res) = (   1, 7.26e-15 )
        linearElasticity: Global solution scaling factor = 1
    Attempt:  0, ConfigurationIter:  0, NewtonIter:  3
        ( Rsolid ) = ( 2.42e-08 )        ( R ) = ( 2.42e-08 )

```
while here is the fully coupled initialization, converges in 1 iteration, no problem:
```
    Attempt:  0, ConfigurationIter:  0, NewtonIter:  0
        ( Rflow ) = ( 0.00e+00 )        ( Rsolid ) = ( 1.69e+01 )        ( Rwell ) = ( 0.00e+00 )        ( R ) = ( 1.69e+01 )
        twoPhaseFlow: Max pressure change: 0.000 Pa (before scaling)
        twoPhaseFlow: Max component density change: 0.000 kg/m3 (before scaling)
        reservoirSystem: Global solution scaling factor = 1
        multiphasePoroelasticity: Max phase volume fraction change: 0.00
    Attempt:  0, ConfigurationIter:  0, NewtonIter:  1
        ( Rflow ) = ( 0.00e+00 )        ( Rsolid ) = ( 6.62e-14 )        ( Rwell ) = ( 0.00e+00 )        ( R ) = ( 6.62e-14 )
        Last LinSolve(iter,res) = (   1, 2.85e-15 )

```

Comment (paveltomin) [reply src/coreComponents/physicsSolvers/solidMechanics/kernels/FixedStressThermoPoromechanics_impl.hpp]: I gave up and simplified it to basic update of bulk density before checking outer loop convergence
That helps to match FIM initialization results

**Pull request merged**